### DISCUSSION TITLE
Rapport PDF personnalisé ?

### DISCUSSION THREAD

**Discussion by Heshzer** [mviewer/mviewer#945] [Q&A]
Bonjour à tous,

Je travaille sur un projet cartographique, et en visitant le site de Rennes Métropole, j'ai découvert une fonctionnalité très intéressante sur leur carte dynamique dédiée au PLUi : https://mviewer.sig.rennesmetropole.fr/?config=apps/PLUi/PLUi.xml#

Lorsqu'on clique sur une parcelle cadastrale ou une section de la carte, un volet s'ouvre à droite (probablement basé sur un template Mustache?). Ce volet permet ensuite de générer un rapport PDF personnalisé avec une vue zoomée de la parcelle et toutes les informations associées à cette zone. Ce n'est pas seulement l'export du visuel de la carte qui m'intéresse, mais bien le fait que le rapport soit généré dynamiquement en fonction de l'endroit où l'on a cliqué.

J'aimerais savoir comment mettre en place ce type de fonctionnalité. Quels outils permettent de récupérer les informations spécifiques à un clic et de les intégrer dans un rapport PDF généré à la volée ? Toute piste technique ou exemple serait vraiment précieux.

Merci beaucoup pour votre aide !

![image](https://github.com/user-attachments/assets/70d6e511-b653-41f7-8383-dc91f1de80bd)
![image](https://github.com/user-attachments/assets/7a921a4e-753f-4a67-8f9b-2976b95a79de)

**Comment by lecault** [comment]
Bonjour,
Contacter Rennes Métropole et @cbriand qui a réalisé ce développement.

**Comment by pbarille** [comment]
Bonjour @Heshzer,  
  
Le Val d'Ille-Aubigné a également développé une fonctionnalité de consultation de PLU basée sur l'outil MViewer, moins avancée que celle de Rennes Métropole cependant : https://geo.valdille-aubigne.fr/mviewer/?config=./apps/plui/plui.xml#
Le tout se base sur une couche parcelle intégrant toutes les informations du PLU par parcelle en format JSON.  
  
Voici le dépôt contenant l'application Mviewer : https://github.com/valdille-aubigne/mviewer-apps/tree/master/plui
Ainsi que la vue permettant de générer la couche parcelle / plu dans postgres :  https://github.com/valdille-aubigne/PLUi/blob/master/vue_parcellaire_opposable.sql  
  
Bonne journée,

**Reply by Gaetanbrl** [reply]
Merci pour ce partage @pbarille ! :)

**Reply by Heshzer** [reply]
Bonjour @pbarille, super je vais regarder ça, merci beaucoup! :)

**Comment by Gaetanbrl** [comment]
Est-ce que certains d'entre vous ont déjà vu des exemples avec l'API carto GPU ?

https://api.gouv.fr/les-api/api_carto_gpu

https://www.geoportail-urbanisme.gouv.fr/
